### ISSUE TITLE
Build failure with `vector` `HEAD`, missing CPP functions

### ISSUE THREAD

**Issue by parsonsmatt** [Bodigrim/bitvec#56]
In my effort to get the ecosystem ready for GHC 9.4, I'm attempting to get [`persistent`](https://github.com/yesodweb/persistent/issues/1398) and dependencies ready. 

`vector` has some changes in `HEAD` that are unreleased, and these changes introduce some build failures. It would appear that `vector.h` changed, and so some things are different.

Build log: 

```
Building library for bitvec-1.1.2.0..
[ 1 of 13] Compiling Data.Bit.Gmp     ( src/Data/Bit/Gmp.hs, dist/build/Data/Bit/Gmp.o, dist/build/Data/Bit/Gmp.dyn_o )
[ 2 of 13] Compiling Data.Bit.PdepPext ( src/Data/Bit/PdepPext.hs, dist/build/Data/Bit/PdepPext.o, dist/build/Data/Bit/PdepPext.dyn_o )
[ 3 of 13] Compiling Data.Bit.Utils   ( src/Data/Bit/Utils.hs, dist/build/Data/Bit/Utils.o, dist/build/Data/Bit/Utils.dyn_o )
[ 4 of 13] Compiling Data.Bit.InternalTS ( src/Data/Bit/InternalTS.hs, dist/build/Data/Bit/InternalTS.o, dist/build/Data/Bit/InternalTS.dyn_o )

src/Data/Bit/Internal.hs:479:3: error:
    Data constructor not in scope:
      BOUNDS_CHECK :: t0 -> String -> Int -> Int -> m0 () -> m ()
    |
479 |   BOUNDS_CHECK(checkIndex) "flipBit" i (MV.length v) $ unsafeFlipBit v i
    |   ^^^^^^^^^^^^

src/Data/Bit/Internal.hs:479:16: error:
    Variable not in scope: checkIndex
    |
479 |   BOUNDS_CHECK(checkIndex) "flipBit" i (MV.length v) $ unsafeFlipBit v i
    |                ^^^^^^^^^^
cabal: Failed to build bitvec-1.1.2.0 (which is required by test:test from
```

**Comment by parsonsmatt** [comment]
Relevant commit: https://github.com/haskell/vector/commit/2197d0a571b47cbab11ab9a7419569e6b87806b3

`BOUNDS_CHECK(checkIndex)` appears to be replaced with `checkIndex Bounds`.

**Comment by Bodigrim** [comment]
@parsonsmatt thanks, but I do not expect `vector` HEAD to be released soon.

**Comment by parsonsmatt** [comment]
Yeah, same. It's necessary for getting GHC 9.4 support rolled out, and I'm trying to help get PRs out to relevant libraries to help on that front. Happy to let this set, maybe in draft state for a while?

Ideally would like to have `persistent` and maybe even `yesod` ready soon, but that touches most of the ecosystem 😅

**Comment by Bodigrim** [comment]
Once `primitive` is unblocked, `vector` / `bitvec` / `vector-algorithms` will be fine with revisions only, I believe.

**Comment by parsonsmatt** [comment]
Ah, I see what you mean - `primitive` works, then `vector` can have a revision and Hackage latest works. 🤔

**Comment by Bodigrim** [comment]
Yep, I made a revision for `vector` and now both `bitvec` and `vector-algorithms` just work. Hope it helps with your endeavour, much appreciated.
